src/rgw/rgw_sal_motr.cc

```diff
@@ -1349,38 +1336,50 @@ int MotrObject::get_obj_attrs(RGWObjectCtx* rctx, optional_yield y, const DoutPr
   if (this->category == RGWObjCategory::MultiMeta)
     return 0;
 
+  bufferlist bl;
   string bname, key;
+  int r = read_obj_attrs(dpp, bname, key, bl, target_obj);
+  if (r < 0) {
+    return r;
+  }
+
+  rgw_bucket_dir_entry ent;
+  bufferlist& blr = bl;
+  auto iter = blr.cbegin();
+  ent.decode(iter);
+  decode(attrs, iter);
+
+  return 0;
+}
+
+void MotrObject::read_bucket_info(std::string& bname, std::string& key, rgw_obj* target_obj)
+{
   if (target_obj) {
     bname = get_bucket_name(target_obj->bucket.tenant, target_obj->bucket.name);
     key   = target_obj->key.to_str();
   } else {
     bname = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
     key   = this->get_key().to_str();
   }
-  ldpp_dout(dpp, 20) << "MotrObject::get_obj_attrs(): "
-                    << bname << "/" << key << dendl;
+}
 
+int MotrObject::read_obj_attrs(const DoutPrefixProvider* dpp, std::string& bname, std::string& key, bufferlist& bl, rgw_obj* target_obj)
+{
+  read_bucket_info(bname, key, target_obj);
+  ldpp_dout(dpp, 20) <<__func__<< ": " << bname << "/" << key << dendl;
```

Comment:
why not to retain log at line 1360 within read_bucket_info() only ?

Reply:
In 1360, it will log only if condition is satisfied. 
We will fetch bname and key in else part if target_obj is not passed in the argument (get_obj_attrs() does not pass target_obj)

Reply:
Am saying, after either 'if' or 'else' case exdcution, bname & key would be there, why cant we log there. Btw line number 1360 is wrt removed code (in red) itself and not new code.
>   ldpp_dout(dpp, 20) << "MotrObject::get_obj_attrs(): "
                    << bname << "/" << key << dendl;

Reply:
alright got it. done the changes